### PR TITLE
feat: add support for stage channels

### PR DESCRIPTION
**Pull request by ayntee**
Ref: https://github.com/discord/discord-api-docs/pull/2751

### CODE REVIEW

File: src/types/channels/channel_types.ts [hunk 1]
```diff
@@ -14,4 +14,6 @@ export enum DiscordChannelTypes {
   GUILD_NEWS,
   /** A channel in which game developers can sell their game on Discord */
   GUILD_STORE,
+  /** A voice channel for hosting events with an audience */
+  GUILD_STAGE_VOICE,
```
Comment (itohatweb) [inline src/types/channels/channel_types.ts:18]:
```suggestion
  GUILD_STAGE_VOICE = 13,
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: src/types/channels/channel_types.ts (hunk 1 above)
Comment (ayntee) [reply src/types/channels/channel_types.ts:18]: Nice catch!

File: src/helpers/guilds/update_bot_voice_state.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *  - current user must already have joined `channel_id`.
+ *  - You must have the `MUTE_MEMBERS` permission to unsuppress yourself. You can always suppress yourself.
+ *  - You must have the `REQUEST_TO_SPEAK` permission to request to speak. You can always clear your own request to speak.
+ *  - You are able to set `request_to_speak_timestamp` to any present or future time.
+ */
+export function updateBotVoiceState(
+  guildId: string,
+  data: UpdateSelfVoiceState,
+) {
+  const payload = camelKeysToSnakeCase(data) as DiscordUpdateSelfVoiceState;
```
Comment (itohatweb) [inline src/helpers/guilds/update_bot_voice_state.ts:22]:
```suggestion
  const payload = camelKeysToSnakeCase<DiscordUpdateSelfVoiceState>(data);
```

File: src/helpers/guilds/update_user_voice_state.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ *  - You must have the `MUTE_MEMBERS` permission. (Since suppression is the only thing that is available currently.)
+ *  - When unsuppressed, non-bot users will have their `request_to_speak_timestamp` set to the current time. Bot users will not.
+ *  - When suppressed, the user will have their `request_to_speak_timestamp` removed.
+ */
+export function updateVoiceState(
+  guildId: string,
+  userId: string,
+  data: UpdateOthersVoiceState,
+) {
+  const payload = camelKeysToSnakeCase(data) as DiscordUpdateOthersVoiceState;
```
Comment (itohatweb) [inline src/helpers/guilds/update_user_voice_state.ts:23]:
```suggestion
  const payload = camelKeysToSnakeCase<DiscordUpdateOthersVoiceState>(data);
```

**Review by itohatweb — CHANGES_REQUESTED** [review]
The `camelKeysToSnakeCase` functions now also take a return type

**Review by itohatweb — APPROVED** [review]

**Pull request merged**